src/mscorlib/src/System/Diagnostics/Stacktrace.cs

```diff
@@ -560,25 +560,37 @@ internal String ToString(TraceFormat traceFormat)
                         sb.Append(']');    
                     }
 
-                    // arguments printing
-                    sb.Append('(');
-                    ParameterInfo[] pi = mb.GetParameters();
-                    bool fFirstParam = true;
-                    for (int j = 0; j < pi.Length; j++)
+#if FEATURE_CORECLR
+                    try
+                    {
+#endif
+                        // arguments printing
+                        ParameterInfo[] pi = mb.GetParameters();
+                        sb.Append('(');
+                        bool fFirstParam = true;
+                        for (int j = 0; j < pi.Length; j++)
+                        {
+                            if (fFirstParam == false)
+                                sb.Append(", ");
+                            else
+                                fFirstParam = false;
+
+                            String typeName = "<UnknownType>";
+                            if (pi[j].ParameterType != null)
+                                typeName = pi[j].ParameterType.Name;
+                            sb.Append(typeName);
+                            sb.Append(' ');
+                            sb.Append(pi[j].Name);
+                        }   
+                        sb.Append(')');
+#if FEATURE_CORECLR
+                    }
+                    catch
                     {
```

Comment:
The failure point I was trying to protect is only the call to `mb.GetParameters()`. The `ParameterType` property just returns a member of the ParameterInfo, so once the `GetParameters` succeeds, there should be no further failure other than OOM. And I don't think we can reliably harden against OOM here.
I'll move the try / catch to make it tight around the `GetParameters`.